test/functional/commands/network-e2e-specs.js

```diff
@@ -1,42 +1,52 @@
-// This currently does not work reliably in CI
-// Further, our CI does not respect skip or @skip-ci
-// investigate and reinstate
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { AndroidDriver } from '../../../..';
+import sampleApps from 'sample-apps';
 
-// import chai from 'chai';
-// import chaiAsPromised from 'chai-as-promised';
-// import { AndroidDriver } from '../../../..';
-// import sampleApps from 'sample-apps';
 
+chai.should();
+chai.use(chaiAsPromised);
 
-// chai.should();
-// chai.use(chaiAsPromised);
+let driver;
+let caps = {
+  app: sampleApps('ApiDemos-debug'),
+  deviceName: 'Android',
+  platformName: 'Android',
+  appActivity: '.view.TextFields'
+};
 
-// let driver;
-// let caps = {
-//   app: sampleApps('ApiDemos-debug'),
-//   deviceName: 'Android',
-//   platformName: 'Android',
-//   appActivity: '.view.TextFields'
-// };
+describe('network connection', function () {
+  this.timeout(180000);
 
-// describe('network connection', function () {
-//   this.timeout(120000);
-//   before(async () => {
-//     driver = new AndroidDriver();
-//     await driver.createSession(caps);
-//   });
-//   after(async () => {
-//     await driver.deleteSession();
-//   });
-//   describe('setNetworkConnection @skip-ci', () => {
-//     function test (value) {
-//       it(`should be able to set to ${value}`, async () => {
-//         await driver.setNetworkConnection(value);
-//         await driver.getNetworkConnection().should.eventually.equal(value);
-//       });
-//     }
-//     for (let value of [1, 2, 4, 6]) {
-//       test(value);
-//     }
-//   });
-// });
+  // these operations get flakier the more times you do them in one
+  // session, so break them out
+  beforeEach(async () => {
+    driver = new AndroidDriver();
+    await driver.createSession(caps);
+  });
+  afterEach(async () => {
+    try {
+      await driver.deleteSession();
+    } catch (err) {
+      console.log('Unable to delete session. Continueing anyway.');
+    }
+  });
+  describe('setNetworkConnection @skip-ci', () => {
+    it('should be able to set to 1', async () => {
+      await driver.setNetworkConnection(1);
+      await driver.getNetworkConnection().should.eventually.equal(1);
```

Comment:
setNetworkConnection calls getNetworkConnection when it's done. You could avoid the additional call. Although, I like the explicit nature of the test.